src/app/Processor/ProcessorDetailPage/ProcessorDetailPage.tsx

```diff
@@ -83,21 +112,87 @@ const ProcessorDetailPage = (): JSX.Element => {
       console.error(processorError);
       goToInstance();
     }
-  }, [bridgeError, processorError, goToHome, goToInstance]);
+    if (updateProcessorError) {
+      console.error(updateProcessorError);
+    }
+  }, [
+    bridgeError,
+    processorError,
+    goToHome,
+    goToInstance,
+    updateProcessorError,
+  ]);
+
+  const processorNotChanged = useCallback(
+    (prevDef: ProcessorResponse, updatedDef: ProcessorRequest): boolean =>
+      isEqualWith(
+        prevDef,
+        updatedDef,
+        (prev: ProcessorResponse, next: ProcessorRequest) =>
+          isEqual(prev.name, next.name) &&
+          isEqual(prev.filters, next.filters) &&
+          isEqual(prev.transformationTemplate, next.transformationTemplate) &&
+          isEqual(prev.action, next.action) &&
+          isEqual(prev.source, next.source)
+      ),
+    []
+  );
+
+  const handleUpdateProcessorSaving = useCallback(
+    (processorRequest: ProcessorRequest) => {
+      if (
+        currentProcessor &&
+        processorNotChanged(currentProcessor, processorRequest)
+      ) {
+        setIsEditing(false);
+        return;
+      }
+      void updateProcessor(instanceId, processorId, processorRequest).then(() =>
```

Comment:
Just asking.. this is a function invocation? Is it customary to include the return type in the invocation?

Reply:
This is a function exported from the hook. It is returning a promise of void.
On that promise I am listening to the success part in order to disable the edit mode when the PUT is executed.
Lint is not complaining, so I guess it is ok from type definition point of view